multiview/embed/mvmds.py

```diff
@@ -0,0 +1,217 @@
+
+                w = np.dot(s2, q)
+
+                w = np.dot(pi, w)
+
+                q = w / np.sqrt(np.dot(w.T, w))
+
+                for yy in np.arange(views):
+
+                    d[:, yy] = np.dot(np.dot(q.T, x[yy]), q)
+
+            Components[:, i] = q[:, 0]
+            pi = pi - np.dot(q, q.T)
+
+        return(Components)
+
+    def fit(self, Xs):
+
+        """
+        Calculates a dimensional reduction based on minimizing Euclidean
+        Distances of multiple views
+
+        Parameters
+        ----------
+
+        Xs: list of array-likes
+                - Xs shape: (n_views,)
+                - xXs[i] shape: (n_samples, n_features_i)
+                The data to fit to. Each sample will receive its own embedding.
+
+        n_components: int (positive), optional, default = 0
+            Number of components requested
+
+        Attributes
+        ----------
+
+        components: A k-dimensional projection of shape [n,k]
+
+        """
+
+        if (self.n_components) > len(Xs[0]):
+            self.n_components = len(Xs[0])
+            warnings.warn('The number of components you have requested is '
+                          + 'greater than the number of samples in the '
+                          + 'dataset. ' + str(self.n_components)
+                          + ' components were computed instead.')
+
+        if (self.n_components) <= 0:
+            raise ValueError('The number of components must be greater than 0 '
+                             + 'and less than the number of features')
+
+        for i in Xs:
+            for j in Xs:
+                if i.shape[0] != j.shape[0]:
+                    raise ValueError('Input matrices must have the same '
+                                     + 'number of samples')
+
+        Xs = check_Xs(Xs, multiview=True)
+
+        mat = np.ones(shape=(len(Xs), len(Xs[0]), len(Xs[0])))
+
+        for i in np.arange(len(Xs)):
+            view = euclidean_distances(Xs[i])
+            view_squared = np.power(np.array(view), 2)
+
+            J = np.eye(len(view)) - (1/len(view))*np.ones(view.shape)
+            B = -(1/2) * np.matmul(np.matmul(J, view_squared), J)
+            mat[i] = B
+
+        self.components = self._cpc(mat)
+
+        return self.components
+
+    def transform(self, Xs):
+
+        """"
+        Embeds data matrix(s) using fitted projection matrices
+
+        Returns back the original Xs. Not needed for implementation
+
+        Parameters
+        ----------
+
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+            The data to embed based on the prior fit function.
+
+        """
+
+        Xs = check_Xs(Xs)
+
+        return Xs
+
+    def fit_transform(self, Xs):
+
+        """"
+        Embeds data matrix(s) using fitted projection matrices
+
+        Parameters
+        ----------
+
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
```

Comment:
change slightly to match https://github.com/NeuroDataDesign/multiview/pull/45/files